src/core/frontend/input.h

```diff
@@ -50,6 +54,15 @@ class InputDevice {
                                [[maybe_unused]] f32 freq_high) const {
         return {};
     }
+    void SetCallback(InputCallback<StatusType> callback_) {
+        callback = callback_;
+    }
+    void TriggerOnChange() {
+        if (callback.on_change) {
+            callback.on_change(std::move(GetStatus()));
```

Comment:
```suggestion
            callback.on_change(GetStatus());
```

Since the result of the call will already be an r-value here.